Lib/tempfile.py

```diff
@@ -868,10 +868,17 @@ def write(self, s):
         return rv
 
     def writelines(self, iterable):
-        file = self._file
-        rv = file.writelines(iterable)
-        self._check(file)
-        return rv
+        if self._max_size == 0 or self._rolled:
+            return self._file.writelines(iterable)
+
+        it = iter(iterable)
+
+        for line in it:
+            self.write(line)
+            if self._rolled:
+                break
+
+        return self._file.writelines(it)
```

Comment:
`self._file.writelines()` is called even if the iterator has been exhausted in the above loop. This is not an issue for normal iterators, except that it is useless. But some iterators (like a file iterator) can yield data after raising StopIteration, so this change can have visible effect. In any case, you can simply move this return in place of `break`.

Reply:
That is a nice way of doing this, thanks! The thought occurred to me but I couldn't think of a good way to catch it.